contrib/kythe/src/python/pants/contrib/kythe/tasks/indexable_java_targets.py

```diff
@@ -0,0 +1,24 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.backend.jvm.targets.jvm_target import JvmTarget
+from pants.build_graph.target_scopes import Scopes
+from pants.util.memo import memoized_method
+
+
+class IndexableJavaTargets(object):
+  """Determines which java targets Kythe should act on."""
+
+  @classmethod
+  @memoized_method
```

Comment:
Hm... I'm not sure what the equality/hashability story of `Context` is, so using memoized_method with it is questionable.

Does this really need to be cached though? One option would be to have the `Indexer` task iterate over all kindex product to collect targets, which would guarantee that you only operated on targets with valid inputs, and would avoid double-filtering the graph.